Lib/Search/EntityRepository.php

```diff
@@ -176,7 +176,15 @@ public function delete($entity, $refresh = false)
      */
     public function deleteByQuery($query)
     {
-        $this->_sm->getClient()->removeAll($this->_class, $query);
+        $queryCriteria = $query->getQuery();
+        $results = $this->scanBy($queryCriteria);
+        foreach ($results as $collection) {
+            foreach ($collection as $entity) {
+                $this->delete($entity, false);
+            }
+            $this->_sm->flush();
+        }
+        $this->_sm->clear();
```

Comment:
I don't really think here it needs "clear". what clear does is simply reset the to persist/ remove queue and touched indices. However, "flush" function above should have ready done this job. indeed, we are calling searchManager->clear() everywhere, but i don't think we need it after flush. @jqin what do you think about that.